Calibration/HcalCalibAlgos/src/AnalyzerMinbias.cc

```diff
@@ -0,0 +1,537 @@
+  myTree->Branch("mom3_Noise",  &mom2_Noise, "mom3_Noise/F");
+  myTree->Branch("mom4_Noise",  &mom4_Noise, "mom4_Noise/F");
+  myTree->Branch("mom0_Diff",   &mom0_Diff,  "mom0_Diff/F");
+  myTree->Branch("mom1_Diff",   &mom1_Diff,  "mom1_Diff/F");
+  myTree->Branch("mom2_Diff",   &mom2_Diff,  "mom2_Diff/F");
+  myTree->Branch("occup",       &occup,      "occup/F");
+  myTree->Branch("trigbit",     &trigbit,    "trigbit/I");
+  myTree->Branch("rnnumber",    &rnnumber,   "rnnumber/D");
+
+  myMap_.clear();
+}
+  
+//  EndJob
+//
+void AnalyzerMinbias::endJob() {
+   
+  int ii=0;
+  for (std::map<std::pair<int,HcalDetId>,myInfo>::const_iterator itr=myMap_.begin(); itr != myMap_.end(); ++itr) {
+    LogDebug("AnalyzerMB") << "Fired trigger bit number " << itr->first.first; 
+    myInfo info = itr->second;
+    if (info.theMB0 > 0) { 
+      mom0_MB    = info.theMB0;
+      mom1_MB    = info.theMB1;
+      mom2_MB    = info.theMB2;
+      mom3_MB    = info.theMB3;
+      mom4_MB    = info.theMB4;
+      mom0_Noise = info.theNS0;
+      mom1_Noise = info.theNS1;
+      mom2_Noise = info.theNS2;
+      mom3_Noise = info.theNS3;
+      mom4_Noise = info.theNS4;
+      mom0_Diff  = info.theDif0;
+      mom1_Diff  = info.theDif1;
+      mom2_Diff  = info.theDif2;
+      rnnumber   = info.runcheck;
+      trigbit    = itr->first.first; 
+      mysubd     = itr->first.second.subdet();
+      depth      = itr->first.second.depth();
+      ieta       = itr->first.second.ieta();
+      iphi       = itr->first.second.iphi();
+      
+      LogDebug("AnalyzerMB") << " Result=  " << trigbit << " " << mysubd
+                             << " " << ieta << " " << iphi << " mom0  "
+                             << mom0_MB << " mom1 " << mom1_MB << " mom2 "
+                             << mom2_MB << " mom3 " << mom3_MB << " mom4 " 
+			     << mom4_MB << " mom0_Noise " << mom0_Noise 
+                             << " mom1_Noise " << mom1_Noise << " mom2_Noise "
+                             << mom2_Noise << " mom3_Noise " << mom3_Noise 
+                             << " mom4_Noise " << mom4_Noise << " mom0_Diff "
+                             << mom0_Diff << " mom1_Diff " << mom1_Diff
+                             << " mom2_Diff " << mom2_Diff;
+      myTree->Fill();
+      ii++;
+    }
+  }
+  cells = ii; 
+  LogDebug("AnalyzerMB") << "cells" << " " << cells;    
+  hOutputFile->Write();   
+  hOutputFile->cd();
+  myTree->Write();
+  for(int i=0; i<4; i++){
+    h_Noise[i]->Write();
+    h_Signal[i]->Write();
+  }
+  hOutputFile->Close() ;
+}
+
+
+//
+// member functions
+//
+  
+// ------------ method called to produce the data  ------------
+  
+void AnalyzerMinbias::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+    
+  rnnum = (float)iEvent.run(); 
+  const HcalRespCorrs* myRecalib=0;
+  if (theRecalib_) {
+    edm::ESHandle <HcalRespCorrs> recalibCorrs;
+    iSetup.get<HcalRespCorrsRcd>().get("recalibrate",recalibCorrs);
```

Comment:
The `HcalRespoCorrsRcd` is not provided via GT (see the `recalibrate` label), but from ASCII file and custom `ESProducer`.